java/vector/src/main/codegen/templates/StructWriters.java

```diff
@@ -184,6 +189,44 @@ public ListWriter list(String name) {
     return writer;
   }
 
+  @Override
+  public MapWriter map(String name) {
+    // returns existing writer
+    final FieldWriter writer = fields.get(handleCase(name));
+    Preconditions.checkNotNull(writer);
```

Comment:
This should be following the paradigm for other complex types that require arguments. In general, `foo(String name)` seems to be used to return an existing writer when the given type requires arguments. The existing writer is expected to have already been created by calling a same-named method with the type's arguments.

For a Map, you need to send in keysSorted for us to be able to construct a writer. There's other types that you need to specify some additional info in order to be able to construct it (Duration needs a TimeUnit, Decimal needs scale / precision). I am guessing that these methods are provided for convenience, so the caller can do:

    writer.decimal256("example", 2, 5).writeDecimal256(..)
    // Now, they can reference that vector later on without constructing a new writer:
    writer.decimal256("example").writeDecimal256(..)

That's why this is returning an existing writer and failing otherwise.

Reply:
We should probably have the writer default to `keysSorted=false` since that will be the most common usage and would only be used if someone really needed this flag. Would that allow you to change this to create a writer by default then, and only supply `keysSorted` optionally?

Reply:
With the default of `keysSorted = false` we can create the writer by default. I've updated this PR with that change.